### ISSUE TITLE
Add support for dynamic locales

### ISSUE THREAD

**Issue by jbphet** [phetsims/greenhouse-effect#204]
PhET has recently decided that we need to support dynamic locales, meaning that the locale can be changed while the sim is running and the strings and layout will correctly update to the new locale.  I asked @kathy-phet whether this feature was needed in Greenhouse Effect, and she said (over Slack):

> Hi [@jbphet], Yes, we will need to switch to dynamic string. It's not something you need to do immediately, but before this sim is published. We've been training [other developers] on the process, to help support, as some of it is fairly manual. JO can also support. So you can take two options - dive in now and see how it works, or wait for kinks to be worked out and convert the sim to use dynamic string a bit down the road.

Then later said:

> Definitely be sure to discuss with @arouinfar and I before beginning the move to dynamic string, so you don't do more work than needed. Thanks.

**Comment by jbphet** [comment]
Here is a link to a parallel issue for another sim.  It may be helpful to reference this in the implementation process.  https://github.com/phetsims/greenhouse-effect/issues/204

**Comment by jbphet** [comment]
This effort is now complete, so I'll close the issue.

For reference, I feel I should describe how I approached converting the a11y strings to StringProperty instances.  In many, though not all, cases the a11y strings are used in "describers" that return a description string when provided with certain information from the model.  For example, the class `ConcentrationDescriber` has a number of static methods, such as `getTimePeriodString`, that provide these descriptive strings.  At first, I considered changing these to produce Property instances that update when the relevant model Properties change *and/or* when the underlying string Properties change.  This proved to be a lot of work, and the API points where these Properties would be used weren't ready for them anyway.  In the end, I took an easier road, and left the API for the describers unchanged and simply used the values of the string Properties to compose the strings that they produce.  This may mean that there will be more work to be done when and if we need to support fully dynamic a11y strings in this sim, but that is likely a ways off, so this seemed like a reasonable path.
